docs/ldml/tr35.md

```diff
@@ -2861,38 +2861,52 @@ Element content whose display may be affected in this way should include an expl
+| `value1`       | <pre>= \[^\\\}\]<br/>\| '\\' quoted</pre>                         | Lm,<br/>\\n,<br/>\\\}                    |
+| `value2`       | <pre>= \[^:\]<br/>\| '\\' quoted</pre>                           | Lm,<br/>\\n,<br/>\\:                    |
+| `element`      | <pre>= char \| string \| var </pre> | %, b, \{hello\}, \{\}, \\x\{61 62\} |
+| `string`      | <pre>= '\{' \(s charInString\)\* s '\}' </pre> | \{hello\} |
+| `char`         | <pre>= \[^ \\^ \\& \\\- \\\[ \\\] \\\\ \\\{ \\$ \[:Pat_WS:\]\]<br/>\| '\\' quoted</pre> | a, b, c, \\n, \\\{, \\$ |
+| `charInString` | <pre>= \[^ \\\\ \\\} \[:Pat_WS:\]\]<br/>\| '\\' quoted</pre> | a, b, c, \\n, \{, $ <br/> *note: only quoted items that resolve to one code point are allowed* | <!-- TODO: Multi escape semantics? If flattening semantics, this note is unnecessary -->
+| `quoted`       | <pre>= 'u' \(hex\{4\} \| bracketedHex\)<br/>\| 'x' \(hex\{2\} \| bracketedHex\)<br/>\| 'U00' \('0' hex\{5\} \| '10' hex\{4\}\)<br/>\| 'N\{' charName '\}'<br/>\| \[\[\\u0000\-\\U00010FFFF\]\-\[uxUN\]\]</pre> | n, U0000FFFE, \{, $, \] <br/> _note: lengths are exact_ |
+| `charName`     | <pre>= s \[A\-Za\-z0\-9\] \[\-A\-Za\-z0\-9\_\\x20\]\* s</pre>                | TIBETAN LETTER \-A                       |
+| `bracketedHex` | <pre>= '\{' s hexCodePoint \(s hexCodePoint\)\* s '\}'</pre>        | \{61 2019 62\}  <br/> *note: the whitespace between hexCodePoints is required here*                         | <!-- TODO: Multi escape semantics? If so, add note explaining here or somewhere else -->
+| `hexCodePoint` | <pre>= hex\{1,5\} \| '10' hex\{4\}</pre>                           |                                         |
+| `hex`          | <pre>= \[0\-9A\-Fa\-f\]</pre>                                       |                                         |
+| `var`          | <pre>= '$' \[:ID_Start:\] \[:ID_Continue:\]\*</pre>                 | $a, $elt5  (optional support)            | <!-- TODO: XID* or ID*? -->
+| `s`            | <pre>= \[:Pattern_White_Space:\]\*</pre>                          | optional whitespace                     |
+
+Some constraints on UnicodeSet syntax are not captured by this EBNF.
+Notably:
+1. Property names and values are restricted to those supported by the implementation, and have additional constraints imposed by [[UAX44](https://www.unicode.org/reports/tr41/#UAX44)].
+2. Quoted characters in strings must resolve to exactly one character. That is, **\[\{\x{61\}\}\]** is valid and equivalent to **\[\{a\}\]**, but **\[\{\\x\{61 62\}\}\]** is invalid and in particular not equivalent to **\[\\x\{61 62\}\]**, which is equivalent to **\[\{ab\}\]**. <!-- TODO: Multi escape semantics? -->
+3. Ranges (**\[X-Y\]**) are only supported in the case that X and Y resolve to single code points. That is, **\[a-b\]** and **\[\{a\}-\{b\}\]** are supported, while **\[a-{bz}\]** and **\[\{ax\}-\{bz\}\]** are not. <!-- TODO: Support for string ranges of length != 1? -->
```

Comment:
maybe 
Ranges (**X-Y**) are only supported in the case that elements X and Y

Reply:
Thank you, changed!